internal/nostr/nostr.go

```diff
@@ -373,32 +408,37 @@ func (svc *Service) handleSubscription(ctx context.Context, subscription *Subscr
 		filter.Until = &until
 	}
 
-	sub, err := relay.Subscribe(ctx, []nostr.Filter{filter})
-	if err != nil {
-		errorChan <- err
-		return
-	}
-	errorChan <- nil
-	go func(){
-		for event := range sub.Events {
-			svc.Logger.WithFields(logrus.Fields{
-				"eventId":   event.ID,
-				"eventKind": event.Kind,
-			}).Infof("received event on subscription %d", subscription.ID)
-			responseEvent := ResponseEvent{
-				SubscriptionId: subscription.ID,
-				NostrId: event.ID,
-				Content: event.Content,
-				RepliedAt: event.CreatedAt.Time(),
+	reconnecting := false
+
+	for {
+		svc.Logger.Infof("subscription %d starting", subscription.ID)
+		sub, err := relay.Subscribe(ctx, []nostr.Filter{filter})
+		if err != nil {
+			if !reconnecting {
```

Comment:
There's a chance the relay is temporarily down the first time you try to subscribe. Does it make sense to not allow subscribing? cc @bumi 

In NWC it will just wait and try again in 10 seconds. It doesn't matter if it's the first try to connect to the relay or not.

Reply:
I think the API should be as much independent of the relay as possible. 
so for API calls that for example only create subscriptions that is fine, then we create the subscription in or DB and try to reconnect to the relay and start the subscription to the relay.

Reply:
Following this.

We now intimate the user as soon as we store the subscription details in our DB that the subscription is successfully created and keep trying every 5s in case of failures.

Reply:
what do you mean by "intimate the user"?

Reply:
As in we send back a Status 200 OK response with subscription uuid and the webhook link like [this](https://guides.getalby.com/developer-guide/v/nostr-wallet-connect-api/building-lightning-apps/nwc-http-api/subscribe-to-events#response)